requestor/core/requestor-api/src/main/java/io/reinert/requestor/uri/UriBuilderImpl.java

```diff
@@ -230,11 +230,11 @@ public Uri build(Object... templateValues) {
         final String parsedFrag = fragment != null ? parsePart(templateValues, templateParams, fragment) : null;
 
         final String[] pathSegments = segments != null ? segments.toArray(new String[segments.size()]) : null;
-        return new Uri(scheme, user, password, host, port, pathSegments, matrixParams, queryParams, parsedFrag);
+        return new UriImpl(scheme, user, password, host, port, pathSegments, matrixParams, queryParams, parsedFrag);
     }
 
     @Override
-    public Uri build(Map<String, ?> values) {
+    public UriImpl build(Map<String, ?> values) {
         if (segments != null) {
             for (int i = 0; i < segments.size(); i++) {
                 final String segment = segments.get(i);
```

Comment:
maybe extracting a private method would be good. How about removing the comment and giving a meaningful name to this new private method?